cli/tools/registry/pm/deps.rs

```diff
@@ -0,0 +1,964 @@
+      Ok(req) => req.req.clone(),
+      Err(err) => {
+        log::warn!("failed to parse package req \"{}\": {err}", value.as_str());
+        continue;
+      }
+    };
+    let alias: &str = key_path.last().unwrap().as_str().trim_end_matches('/');
+    let alias = (alias != req.name).then(|| alias.to_string());
+    if !filter.should_include(alias.as_deref(), &req, kind) {
+      continue;
+    }
+    let id = DepId(deps.len());
+    deps.push(Dep {
+      location: DepLocation::DenoJson(
+        deno_json.clone(),
+        key_path,
+        import_map_kind,
+      ),
+      kind,
+      req,
+      id,
+      alias,
+    });
+  }
+}
+
+fn add_deps_from_package_json(
+  package_json: &PackageJsonRc,
+  mut filter: impl DepFilter,
+  deps: &mut Vec<Dep>,
+) {
+  let package_json_deps = resolve_local_package_json_deps(package_json);
+  for (k, v) in package_json_deps {
+    let (package_dep_kind, v) = match v {
+      Ok((k, v)) => (k, v),
+      Err(e) => {
+        log::warn!("bad package json dep value: {e}");
+        continue;
+      }
+    };
+    match v {
+      deno_package_json::PackageJsonDepValue::Req(req) => {
+        let alias = k.as_str();
+        let alias = (alias != req.name).then(|| alias.to_string());
+        if !filter.should_include(alias.as_deref(), &req, DepKind::Npm) {
+          continue;
+        }
+        let id = DepId(deps.len());
+        deps.push(Dep {
+          id,
+          kind: DepKind::Npm,
+          location: DepLocation::PackageJson(
+            package_json.clone(),
+            KeyPath::from_parts([package_dep_kind.into(), k.into()]),
+          ),
+          req,
+          alias,
+        })
+      }
+      deno_package_json::PackageJsonDepValue::Workspace(_) => continue,
+    }
+  }
+}
+
+fn deps_from_workspace(
+  workspace: &Arc<Workspace>,
+  dep_filter: impl DepFilter,
+) -> Result<Vec<Dep>, AnyError> {
+  let mut deps = Vec::with_capacity(256);
+  for deno_json in workspace.deno_jsons() {
+    add_deps_from_deno_json(deno_json, dep_filter, &mut deps);
+  }
+  for package_json in workspace.package_jsons() {
+    add_deps_from_package_json(package_json, dep_filter, &mut deps);
+  }
+
+  Ok(deps)
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct DepId(usize);
+
+#[derive(Debug, Clone)]
+pub enum Change {
+  Update(DepId, VersionReq),
+}
+
+pub trait DepFilter: Copy {
+  fn should_include(
+    &mut self,
+    alias: Option<&str>,
+    package_req: &PackageReq,
+    dep_kind: DepKind,
+  ) -> bool;
+}
+
+impl<T> DepFilter for T
+where
+  T: FnMut(Option<&str>, &PackageReq, DepKind) -> bool + Copy,
+{
+  fn should_include<'a>(
+    &mut self,
+    alias: Option<&'a str>,
+    package_req: &'a PackageReq,
+    dep_kind: DepKind,
+  ) -> bool {
+    (*self)(alias, package_req, dep_kind)
+  }
+}
+
+#[derive(Clone, Debug)]
+pub struct PackageLatestVersion {
+  pub semver_compatible: Option<PackageNv>,
+  pub latest: Option<PackageNv>,
+}
+
+pub struct DepManager {
```

Comment:
Maybe `DepUpdater`? Generally `Manager` is a term that's avoided because it means the struct has more than one responsibility.